File: Solutions/Neetcode/Trees/2-MaxDepthBinaryTree/solution.py
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """

        if not root:
            return 0

        max_left = self.maxDepth(root.left)
        max_right = self.maxDepth(root.right)

        return max(max_left, max_right) + 1

File: Solutions/Neetcode/Trees/2-MaxDepthBinaryTree/solution.ts
import { TreeNode } from "../../../../Types/TreeNode/TreeNode";

export function maxDepth(root: TreeNode | null): number {
  if (!root) return 0;

  const maxLeft = maxDepth(root.left);
  const maxRight = maxDepth(root.right);

  return Math.max(maxLeft, maxRight) + 1;
}
